src/storage/metric/transaction_metric.cpp

```diff
@@ -0,0 +1,68 @@
+#include "storage/metric/transaction_metric.h"
+#include <vector>
+#include "catalog/catalog_defs.h"
+#include "storage/metric/thread_level_stats_collector.h"
+#include "transaction/transaction_manager.h"
+#include "util/transaction_test_util.h"
+
+namespace terrier::storage::metric {
+
+catalog::SqlTableHelper *TransactionMetricRawData::GetStatsTable(transaction::TransactionManager *const txn_manager,
+                                                                 catalog::Catalog *const catalog,
+                                                                 transaction::TransactionContext *txn) {
+  const catalog::db_oid_t terrier_oid(catalog::DEFAULT_DATABASE_OID);
+  auto db_handle = catalog->GetDatabaseHandle();
+  auto table_handle = db_handle.GetNamespaceHandle(txn, terrier_oid).GetTableHandle(txn, "public");
+
+  // define schema
+  std::vector<catalog::Schema::Column> cols;
+  cols.emplace_back("id", type::TypeId::BIGINT, false, catalog::col_oid_t(catalog->GetNextOid()));
+  cols.emplace_back("latency", type::TypeId::BIGINT, false, catalog::col_oid_t(catalog->GetNextOid()));
+  cols.emplace_back("tuple_read", type::TypeId::BIGINT, false, catalog::col_oid_t(catalog->GetNextOid()));
+  cols.emplace_back("tuple_insert", type::TypeId::BIGINT, false, catalog::col_oid_t(catalog->GetNextOid()));
+  cols.emplace_back("tuple_delete", type::TypeId::BIGINT, false, catalog::col_oid_t(catalog->GetNextOid()));
+  cols.emplace_back("tuple_update", type::TypeId::BIGINT, false, catalog::col_oid_t(catalog->GetNextOid()));
+  catalog::Schema schema(cols);
+
+  // create table
+  auto table_ptr = table_handle.GetTable(txn, "txn_metric_table");
+  if (table_ptr == nullptr) table_ptr = table_handle.CreateTable(txn, schema, "txn_metric_table");
+  return table_ptr;
+}
+
+void TransactionMetricRawData::UpdateAndPersist(transaction::TransactionManager *const txn_manager,
+                                                catalog::Catalog *catalog, transaction::TransactionContext *txn) {
+  auto table = GetStatsTable(txn_manager, catalog, txn);
+  TERRIER_ASSERT(table != nullptr, "Stats table cannot be nullptr.");
+  std::vector<type::TransientValue> row;
+
+  for (auto &entry : data_) {
+    // one iteration per transaction
+    auto txn_oid = static_cast<uint64_t>(entry.first);
+    auto &counter = entry.second;
+    auto latency = counter.latency_;
+    auto tuple_read = counter.tuple_read_;
```

Comment:
This (and similarly for the other `counter` attributes) is a temporary variable only being used in one place in an `if` block ([here](https://github.com/cmu-db/terrier/pull/338/files#diff-e6e2930861c783ee7a094b005eaa7531R56)). Wouldn't it be better to use these directly in the `if` block with scope only in that block?